shaarli_client/utils.py

```diff
@@ -24,26 +24,22 @@ def generate_all_endpoints_parsers(subparsers, endpoints):
         generate_endpoint_parser(subparsers, ep_name, ep_metadata)
 
 
-def write_output(args, response):
-    """Write the API response to file or standard output"""
-    if not args.outfile:
-        if args.format == 'json':
-            print(response.json())
-        elif args.format == 'pprint':
-            print(json.dumps(response.json(), sort_keys=True, indent=4))
-        elif args.format == 'text':
-            print(response.text)
+def format_response(output_format, response):
+    """Format the API response to the desired output format"""
+    if output_format == 'json':
+        return response.json()
+    elif output_format == 'pprint':
+        return json.dumps(response.json(), sort_keys=True, indent=4)
+    elif output_format == 'text':
+        return response.text
     else:
-        try:
-            output_file = open(args.outfile, 'w+')
-        except:
-            raise OSError("File %s cannot be written" % args.outfile)
-        if args.format == 'json':
-            output_file.write(response.json())
-        elif args.format == 'pprint':
-            output_file.write(
-                json.dumps(response.json(), sort_keys=True, indent=4)
-            )
-        elif args.format == 'text':
-            output_file.write(response.text)
-        output_file.close()
+        raise ValueError("%s is not a supported format." % output_format)
+
+def output_to_file(output_filename, output):
+    """Write the program output to a file"""
+    try:
+        output_file = open(output_filename, 'w+')
```

Comment:
1. `w+` should be avoided here, as it will produce invalid documents when using a JSON-formatted output - there'll be a missing comma delimiter between objects

2. the preferred way for handling files is by using a context manager:

```python
with open('myfile', 'w') as myfile_handler:
    myfile_handler.write("Plop!")
```

with exception handling:

```python
try:
    with open('myfile', 'w') as myfile_handler:
        myfile_handler.write("Plop!")
except OSError as exc:
    # process the exception
```